Symbol instantiation causes panic/UB

Following code fails to run on latest version:
```rust
use string_interner::{symbol::SymbolU16, Symbol};
fn main() {
	let s = SymbolU16::try_from_usize(std::u16::MAX as usize);
	dbg!(s);
}
```
It gives the following error:
```
thread 'main' panicked at 'attempt to add with overflow', .cargo/registry/src/github.com-1ecc6299db9ec823/string-interner-0.10.0/src/symbol.rs:64:77
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.46/src/backtrace/libunwind.rs:86
   1: backtrace::backtrace::trace_unsynchronized
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.46/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print_fmt
             at src/libstd/sys_common/backtrace.rs:78
   3: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
             at src/libstd/sys_common/backtrace.rs:59
   4: core::fmt::write
             at src/libcore/fmt/mod.rs:1069
   5: std::io::Write::write_fmt
             at src/libstd/io/mod.rs:1504
   6: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:62
   7: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:49
   8: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:198
   9: std::panicking::default_hook
             at src/libstd/panicking.rs:218
  10: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:511
  11: rust_begin_unwind
             at src/libstd/panicking.rs:419
  12: core::panicking::panic_fmt
             at src/libcore/panicking.rs:111
  13: core::panicking::panic
             at src/libcore/panicking.rs:54
  14: <string_interner::symbol::SymbolU16 as string_interner::symbol::Symbol>::try_from_usize
             at .cargo/registry/src/github.com-1ecc6299db9ec823/string-interner-0.10.0/src/symbol.rs:64
```

That is because
https://github.com/Robbepop/string-interner/blob/85b1829b55496308bdc8bc100fe3eb2daf73e0d6/src/symbol.rs#L60-L68
is invalid in the `u16` case. (And I would guess it's Undefined Behaviour in release mode, as it calls `new_unchecked` on a non-zero type with a zero value).

I beleive the proper way to implement this, without `unsafe`, with the same number of instructions, and simpler, would be:
```rust
            fn try_from_usize(index: usize) -> Option<Self> {
                 <$non_zero>::new((index as $base_ty).wrapping_add(1))
            }
```

Also, `usize::MAX` only exists in very recent versions of Rust, so this would increase compatibility (otherwise triggers the following error:)
```
    Updating crates.io index
   Compiling string-interner v0.10.0
error[E0599]: no associated item named `MAX` found for type `usize` in the current scope
  --> .cargo/registry/src/github.com-1ecc6299db9ec823/string-interner-0.10.0/src/symbol.rs:62:35
   |
62 |                   if index < usize::MAX {
   |                                     ^^^ associated item not found in `usize`
...
77 | / gen_symbol_for!(
78 | |     /// Symbol that is 16-bit in size.
79 | |     ///
80 | |     /// Is space-optimized for used in `Option`.
81 | |     struct SymbolU16(NonZeroU16; u16);
82 | | );
   | |__- in this macro invocation
   |
help: you are looking for the module in `std`, not the primitive type
   |
62 |                 if index < std::usize::MAX {
   |                            ^^^^^^^^^^^^^^^
error[E0599]: no associated item named `MAX` found for type `usize` in the current scope
  --> .cargo/registry/src/github.com-1ecc6299db9ec823/string-interner-0.10.0/src/symbol.rs:62:35
   |
62 |                   if index < usize::MAX {
   |                                     ^^^ associated item not found in `usize`
...
83 | / gen_symbol_for!(
84 | |     /// Symbol that is 32-bit in size.
85 | |     ///
86 | |     /// Is space-optimized for used in `Option`.
87 | |     struct SymbolU32(NonZeroU32; u32);
88 | | );
   | |__- in this macro invocation
   |
help: you are looking for the module in `std`, not the primitive type
   |
62 |                 if index < std::usize::MAX {
   |                            ^^^^^^^^^^^^^^^

error[E0599]: no associated item named `MAX` found for type `usize` in the current scope
  --> .cargo/registry/src/github.com-1ecc6299db9ec823/string-interner-0.10.0/src/symbol.rs:62:35
   |
62 |                   if index < usize::MAX {
   |                                     ^^^ associated item not found in `usize`
...
89 | / gen_symbol_for!(
90 | |     /// Symbol that is the same size as a pointer (`usize`).
91 | |     ///
92 | |     /// Is space-optimized for used in `Option`.
93 | |     struct SymbolUsize(NonZeroUsize; usize);
94 | | );
   | |__- in this macro invocation
   |
help: you are looking for the module in `std`, not the primitive type
   |
62 |                 if index < std::usize::MAX {
   |                            ^^^^^^^^^^^^^^^

error: aborting due to 3 previous errors
```

--- a/src/symbol.rs
+++ b/src/symbol.rs
@@ -60,14 +60,8 @@ macro_rules! gen_symbol_for {
         impl Symbol for $name {
             #[inline]
             fn try_from_usize(index: usize) -> Option<Self> {
-                if index < <$base_ty>::MAX as usize {
-                    return Some(Self {
-                        // SAFETY: This is safe because we have already checked
-                        //         above that `index + 1` won't result in 0.
-                        value: unsafe { <$non_zero>::new_unchecked(index as $base_ty + 1) },
-                    })
-                }
-                None
+                <$non_zero>::new((index as $base_ty).wrapping_add(1))
+                    .map(|non_zero| Self { value: non_zero })
             }
 
             #[inline]

